### PR TITLE
[prepare-release] Use fullpath and do not forward stdout/stderr to /dev/null

### PR DESCRIPTION
**Pull request by vpnachev**
**How to categorize this PR?**

/area delivery dev-productivity
/kind bug

**What this PR does / why we need it**:
Use fullpath and do not forward stdout/stderr to /dev/null

Currently there are issues releasing some of the extensions and they are failing with something like 
```
...
cp: cannot stat './.ci/..': No such file or directory
...
```

The reason for this is that the current dir is change here https://github.com/gardener/gardener/blob/278e9b4e584404e14ec1b0d214987a19f4352eb0/hack/.ci/prepare_release#L49 and then the relative path is no longer valid. 

Also, now it does not hide the logs/errors from make.bash from the golang installation.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```developer bugfix

```

/cla

### CODE REVIEW

**Comment by vpnachev** [comment]
/hold

I am evaluating different approach for golang installation.

**Comment by vpnachev** [comment]
/unhold

I will open another PR for the golang installation improvement.
/milestone v1.26

File: hack/.ci/prepare_release [hunk 1]
```diff
@@ -16,7 +16,7 @@
 
 set -e
 
-repo_root_dir="$1"
+repo_root_dir="$(realpath $1)"
```
Comment (ialidzhikov) [inline hack/.ci/prepare_release:19]: I think that the `cp` failure that you described occurs only when prepare_release script is invoked from relative path:

```
# ./.ci/prepare_release

cp: cannot stat './.ci/..': No such file or directory
```

but 

```
# /mnt/.ci/prepare_release
```

succeeds.

In general, it makes sense to support both. 👍

**Review by ialidzhikov — APPROVED** [review]
/lgtm

**Pull request merged**